### PR TITLE
[fwutil]: Add firmware management utility

### PR DESCRIPTION
**Pull request by mudsut4ke**
**- What I did**

- Add firmware management CLI
- Add support for querying module's firmware version
- Add support for Install firmware image to module

**- New command syntax:**
```
root@sonic:~# fwutil show version
root@sonic:~# fwutil install -m <module name> <path/url>
```
**- Example of command output**
```
root@sonic:~# fwutil show version
Module    Submodule    Version
--------  -----------  ---------
BIOS                   5.6.5
CPLD      CPLD4        4.4
CPLD      CPLD5        5.4
CPLD      CPLD2        2.5
CPLD      CPLD3        3.4
CPLD      CPLD1        1.4

root@sonic:~# fwutil install -m CPLD http://192.168.20.106/slx/Seastone_CPLD_V0_7.vme
Downloading image...
Command: ispvm /tmp/fw_image.vme

                 Lattice Semiconductor Corp.

             ispVME(tm) V12.2 Copyright 1998-2011.

For daisy chain programming of all in-system programmable devices

Processing virtual machine file (/tmp/fw_image.vme)......

CREATED BY: ispVM(R) System Version 18.0.1
CREATION DATE: 12/06/18 16:03:31  

+=======+
| PASS! |
+=======+

Done

```

Signed-off-by: Wirut Getbamrung wgetbumr@celestica.com

### CODE REVIEW

File: fwutil/main.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+                                PLATFORM_SPECIFIC_MODULE_NAME + ".py"])
+        module = imp.load_source(PLATFORM_SPECIFIC_MODULE_NAME, module_file)
+    except IOError, e:
+        log_error("Failed to load platform module '%s': %s" % (
+            PLATFORM_SPECIFIC_MODULE_NAME, str(e)), True)
+        return -1
+
+    try:
+        platform_fwutil_class = getattr(module, PLATFORM_SPECIFIC_CLASS_NAME)
+        platform_fwutil = platform_fwutil_class()
```
Comment (lguohan) [inline fwutil/main.py]: where does this fwutil class defined?

File: fwutil/main.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+        proc.wait()
+        hwsku = stdout.rstrip('\n')
+    except OSError, e:
+        raise OSError("Cannot detect platform")
+
+    return (platform, hwsku)
+
+
+# Loads platform specific fwutil module from source
+def load_platform_fwutil():
```
Comment (lguohan) [inline fwutil/main.py]: can you add this platform v2 api for this in the sonic-platform-common api?

File: fwutil/main.py (hunk 1 above)
Comment (mudsut4ke) [reply fwutil/main.py]: This class is defined on device plugin:
https://github.com/Azure/sonic-buildimage/pull/2824

File: fwutil/main.py (hunk 2 above)
Comment (mudsut4ke) [reply fwutil/main.py]: OK, I will do it

Comment (mudsut4ke) [reply fwutil/main.py]: Done (https://github.com/Azure/sonic-platform-common/pull/30)

**Comment by mudsut4ke** [comment]
@lguohan , please help review another PR for this feature 
- https://github.com/Azure/sonic-platform-common/pull/30
- https://github.com/Azure/sonic-buildimage/pull/2824

**Comment by keboliu** [comment]
@mudsut4ke if my understanding is correct, for now, this utility is only applicable to Celestica platform? If so I would like to suggest adding this utility as platform-specific. Currently, we have a cli group "platform" under "config" command, you may add a subgroup for your platform and make these subgroup only exposed on your platform.

**Comment by mudsut4ke** [comment]
@jleveque Can you review this design ?

**Pull request closed without merging**